reVX/least_cost_xmission/layers/friction_barrier_builder.py

```diff
@@ -0,0 +1,347 @@
+            return processed
+
+    def _process_vector_layer(self, fname: str, config: FBLayerConfig
+                              ) -> npt.NDArray:
+        """
+        Rasterize a vector layer
+
+        Parameters
+        ----------
+        fname : str
+            Name of vector layer to rasterize
+        config : FBLayerConfig
+            Config for layer
+
+        Returns
+        -------
+        array-like
+            Rasterized vector
+        """
+        if config.rasterize is None:
+            raise ValueError(f'{fname} is a vector but the config is missing '
+                             f'key "rasterize": {config}')
+
+        r_config = config.rasterize
+
+        temp = rasterize(fname, self._io_handler.profile,
+                         buffer_dist=r_config.buffer,
+                         burn_value=r_config.value,
+                         dtype=self._dtype,
+                         reproject_vector=r_config.reproject)
+
+        if config.extent == ALL:
+            return temp
+
+        mask = self.__get_mask(config.extent)
+        processed = np.zeros(self._io_handler.shape, dtype=self._dtype)
+        processed[mask] = temp[mask]
+        return processed
+
+    def _process_forced_inclusions(self, data: npt.NDArray,
+                                   fi_layers: Dict[str, FBLayerConfig]
+                                   ) -> npt.NDArray:
+        """
+        Use forced inclusion (FI) layers to remove barriers/friction. Any
+        value > 0 in the FI layers will result in a 0 in the corresponding
+        cell in the returned raster.
+
+        Parameters
+        ----------
+        data : array-like
+            Composite friction or barrier layer
+        fi_layers : dict
+            Dict of forced inclusions layers keyed by GeoTIFF filename.
+
+        Returns
+        -------
+        array-like
+            Composite layer with forced inclusions added
+        """
+        fi = np.zeros(self._io_handler.shape)
+
+        for fname, config in fi_layers.items():
+            if Path(fname).suffix.lower() not in ['.tif', '.tiff']:
+                logger.error(f'Forced inclusion file {fname} does not end '
+                             'with .tif. GeoTIFFs are the only format '
+                             'allowed for forced inclusions.')
+            map_given = config.map is not None
+            range_given = config.range is not None
+            rasterize_given = config.rasterize is not None
+            if map_given or range_given or rasterize_given:
+                logger.error('`map`, `range`, and `rasterize` are not '
+                             'allowed if `forced_inclusion` is True, but '
+                             f'one was found in config: {fname}: {config}')
+
+            # Past guard clauses, process FI
+            if config.extent != ALL:
+                mask = self.__get_mask(config.extent)
+
+            temp = self._io_handler.load_tiff(fname)  # TODO allow reprojecting
```

Comment:
I forced reproject to True. I was thinking it would be nice to default to not projecting unless the user requests it on a per layer basis. That forces them to understand their data a little better. I'm second guessing whether or not that's the best approach and decided to go with the easy route.